liberty-managed/src/main/java/org/jboss/arquillian/container/was/wlp_managed_8_5/WLPManagedContainer.java

```diff
@@ -1201,6 +1218,343 @@ private void throwDeploymentException(String applicationName) throws DeploymentE
+                       xCon = clazz.getConstructor(String.class);
+                       thisServerThrowable = (Throwable) xCon.newInstance(new Object[] { x.logMsg });
+                       thisServerThrowable.initCause(cause);
+                       
+                   } catch (NoSuchMethodException e2) {
+                       try {
+                           // We are prepared to loose the log line to get the correct exception
+                           xCon = clazz.getConstructor(Throwable.class);
+                           thisServerThrowable = (Throwable) xCon.newInstance(new Object[] { cause });
+                           
+                       } catch (NoSuchMethodException e3) {
+                           // We could not get the right result;
+                           log.warning("Arquillian container could not load external Throwable 2, from: " + x);
+                           thisServerThrowable = new UnloadableLogError(x.logMsg, cause);
+                       }
+                   }
+               }
+
+           }
+
+       } catch (InstantiationException e) {
+           log.warning(e.getMessage() + "Arquillian container could not load external Throwable 3, from: " + x);
+       } catch (IllegalAccessException e) {
+           log.warning(e.getMessage() + "Arquillian container could not load external Throwable 4, from: " + x);
+       } catch (IllegalArgumentException e) {
+           log.warning(e.getMessage() + "Arquillian container could not load external Throwable 5, from: " + x);
+       } catch (InvocationTargetException e) {
+           log.warning(e.getMessage() + "Arquillian container could not load external Throwable 6, from: " + x);
+       }
+
+       if( thisServerThrowable == null ) {
+           thisServerThrowable = new UnloadableLogError(x.logMsg, cause);
+       }
+       
+       log.finest( "Actually created class= " + thisServerThrowable.getClass().getName() + " instance msg=" + thisServerThrowable.getMessage() );
+       
+       return thisServerThrowable;
+   }
+
+   /**
+    * Get a list of ExMsg objects representing Exceptions in a FFDC log file
+    * 
+    * @param ffdc the ffdc file being processed
+    * @return an ordered list of exceptions, as ExMsg, foundß
+    */
+   private ArrayList<ExMsg> getExceptionCausalChain(File ffdc)  {
+
+       ArrayList<ExMsg> result = new ArrayList<ExMsg>();
+
+       BufferedReader br = null;
+       try {
+           br = new BufferedReader(new InputStreamReader(new FileInputStream(ffdc)));
+           String line;
+           // FFDC files have the form:
+           // Stack Dump = com.ibm.ws.container.service.state.StateChangeException:
+           // Caused by: org.jboss.weld.exceptions.WeldException:
+           // Caused by: org.jboss.weld.exceptions.DefinitionException:
+           while ((line = br.readLine()) != null) {
+               Pattern p = Pattern.compile("(Stack Dump = |Caused by: )([a-zA-Z\\.]*?):(.*)");
```

Comment:
This regex doesn't match all valid characters in a class name. Numbers, underscores, dollar signs and digits are also valid in a java identifier. [JLS reference](https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.8)

It should also probably use `+?` instead of `*?` to ensure we never get back a zero length class name (shouldn't happen unless the file format changes though).

We could also go the other way and match `[^:]+?` if we know the colon character is always going to be there.

Reply:
Have done this:
final String JAVA_ID_INITIAL_CHAR = "[\\p{Letter}_$]";  //First Character in Java Id Name
           final String JAVA_ID_NON_INITIAL_CHAR = "\\p{Letter}\\d_$]";
           final String JAVA_ID_TAIL = JAVA_ID_NON_INITIAL_CHAR+"*";
           final String JAVA_ID = JAVA_ID_INITIAL_CHAR + JAVA_ID_TAIL;
           final String JAVA_PACKAGE_QUALIFIER = "("+JAVA_ID+"\\.)*";
           final String JAVA_QUALIFIED_CLASS = JAVA_PACKAGE_QUALIFIER + JAVA_ID;
           // ([\\p{Letter}_$][\\p{Letter}\\d_$]*\\.)*[\\p{Letter}_$][a-zA-Z\\d_$]*
           
           while ((line = br.readLine()) != null) { 
        	       Pattern p = Pattern.compile("(Stack Dump = |Caused by: )(" + JAVA_QUALIFIED_CLASS + "):(.*)");